### DISCUSSION TITLE
Why txindex is needed? in bitcoin.conf

### DISCUSSION THREAD

**Discussion by FranciscoOssian** [ACINQ/eclair#2216] [Q&A] [answered]
I was thinking i could use pruning mode, but found that prune are opposite of txindex=1 and Eclair needs txindex=1

So, I want to understand why the whole blockchain is needed. pls. just curiosity

**Comment by sstone** [comment, marked as answer]
LN nodes need to watch the blokchain to validate channels, and monitor their own channels and react when their funding transactions are spent. This requires indexing the blockchain.
There are basically 2 options:
- rely on bitcoin core to to the indexing for you
- download and index blocks yourself.

We've chosen to rely on bitcoin core a lot, for transaction indexes but also to manage funds (we use it as a wallet) and bump transactions. It has many advantages (we benefit from bitcoin core's funding and fee management capabilities, we're "stateless" wrt to the bitcoin blockchain) and drawbacks, such as requiring -txindex which is not compatible with -pruned.

One thing to keep in mind is that eclair was designed form the beginning with reliability and scalability in mind and it can easily manage 1000s of channels on a single standard server. For this use case requiring -txindex is not a huge constraint.
